docs/proposals/toXml_fromXml_apis.md

```diff
@@ -0,0 +1,129 @@
+# Proposal: Introduce APIs to perform conversions between the Map/Ballerina record and XML
+
+_Owners_: @daneshk @kalaiyarasiganeshalingam  
+_Reviewers_: @daneshk  
+_Created_: 2022/06/17   
+_Updated_: 2022/06/17  
+_Issues_: [#2819](https://github.com/ballerina-platform/ballerina-standard-library/issues/2819)
+
+## Summary
+The Ballerina Xmldata module doesn't have any API to convert the Ballerina Record/Map to XML and XML to map. 
+So, this proposal introduces a new `fromXml` API to convert the XML data to Record/Map and `toXml` API to convert the Record/Map to XML.
+
+## Goals
+Provide a way to perform conversions between XML and Record/Map.
+
+## Motivation
+At the moment, users have to write their own custom implementation to perform conversions between Ballerina records/Map to XML.
+Therefore, It would be easy for them if we provided APIs to convert.
+
+Note: This feature(fromRecord()) is also required by the connector team. When they write a connector for the SOAP backend service(e.g Netsuite Connector),
+they need to convert the Ballerina record value to XML payload. As mentioned in the summary, we don't have a way to do this conversion. 
+They are planning to remove their custom implementation and use `xmldata` module, once we have this feature. As per the [discussion](https://github.com/ballerina-platform/ballerina-standard-library/issues/2819), 
+we are planning to introduce `fromXml` API for their requirement which is an improved API from `fromRecord`. 
+This will be used to do the conversion of the Ballerina record|map data to XML.
+
+## Description
+
+### APIs definition:
+
+```ballerina
+# Converts an XML to its `Map` or `Record` representation.
+# The namespaces and attributes will not be considered a special case.
```

Comment:
what does this mean?

Reply:
`fromXml` API considers namespace and attribute and treats them as special cases, but this API doesn't treat them as special cases.